`bson` headers can't be found when compiling from source and using a system version of `libbson`

When compiling from source and using a system version of `libbson` also compiled from source, I get: `pymongoarrow/lib.cpp:1457:10: fatal error: 'bson/bson.h' file not found`

If you're not compiling pymongoarrow from source, it by default installs `libbson` to a [non standard location](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/advanced-installation/#installing-to-non-standard-directories), so I've created a fake directory here `/usr/local/lib/libbson`, created symlinks to this directory, and modified `pkg-config` to point to this directory.

```bash
Using LIBBSON_INSTALL_DIR: /usr/local/lib/libbson
Using lib_dir: ['/usr/local/lib/libbson/lib']
Using lib_file: /usr/local/lib/libbson/lib/libbson-1.0.so.0
Copying /usr/local/lib/libbson/lib/libbson-1.0.so.0 to /tmp/mongo-arrow-1.9.0/bindings/python/pymongoarrow
bundled SONAME + headers into /tmp/mongo-arrow-1.9.0/bindings/python/pymongoarrow
cflags: -I/usr/local/lib/libbson/include/libbson-1.0
ldflags: -L/usr/local/lib/libbson/lib -lbson-1.0
```

And still receiving the error. Linking https://github.com/mongodb-labs/mongo-arrow/issues/326 as it may be related.

--- a/.github/workflows/test-python.yml
+++ b/.github/workflows/test-python.yml
@@ -94,6 +94,36 @@ jobs:
         env:
           UV_PYTHON: ${{matrix.python-version}}
         run: just test-no-optional
+
+  bson2:
+    name: bson2
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - uses: actions/setup-python@v5
+      - name: Install Deps
+        run: |
+          python -m pip install rust-just uv
+      - name: Set up env
+        run: |
+           echo "LIBBSON_INSTALL_DIR=$PWD/libbson" >> $GITHUB_ENV
+           echo "LD_LIBRARY_PATH=$PWD/libbson/lib" >> $GITHUB_ENV
+           echo "LIBBSON_VERSION=2.0.1" >> $GITHUB_ENV
+      - name: Install libbson
+        run: just build-libbson
+      - name: Ensure imports with no test deps
+        run: just import-check
+      - name: Start MongoDB
+        if: ${{ startsWith(runner.os, 'Linux') }}
+        uses: supercharge/mongodb-github-action@90004df786821b6308fb02299e5835d0dae05d0d # 1.12.0
+        with:
+          mongodb-version: 4.4
+          mongodb-replica-set: test-rs
+      - name: Run the tests
+        run: just test
   docs:
     runs-on: ubuntu-latest
     steps:

--- a/bindings/python/CONTRIBUTING.md
+++ b/bindings/python/CONTRIBUTING.md
@@ -186,6 +186,10 @@ $ conda install --channel conda-forge libbson pkg-config
 The minimum required version is listed in `pymongoarrow/version.py`. If
 you try to build with a lower version a `ValueError` will be raised.
 
+Our minimum supported major version is 1.x, and that is what we include in our wheels.
+In order to build with `bson2`, you can `export LIBBSON_VERSION=2.<minor>.<patch>` before
+running `just build-libbson` or building the library.
+
 ## Build
 
 Typically we will use the provided `just` scripts and will not build

--- a/bindings/python/setup.py
+++ b/bindings/python/setup.py
@@ -88,7 +88,7 @@ def append_libbson_flags(module):
                 lib_path = os.path.join(lib_dir, bson_lib).replace(os.sep, "/")
                 if os.path.exists(lib_path):
                     module.extra_link_args = [lib_path]
-                    include_path = bson_lib.replace(".lib", "")
+                    include_path = "libbson-1.0" if BSON_MAJOR_VERSION == 1 else "bson2"
                     include_dir = os.path.join(install_dir, "include", include_path).replace(
                         os.sep, "/"
                     )

--- a/bindings/python/setup.py
+++ b/bindings/python/setup.py
@@ -18,6 +18,8 @@ COPY_LIBBSON = not os.environ.get("MONGO_NO_COPY_LIBBSON", "")
 # Whether to create libarrow symlinks on posix systems.
 CREATE_LIBARROW_SYMLINKS = os.environ.get("MONGO_CREATE_LIBARROW_SYMLINKS", "1")
 
+LIBBSON_VERSION = os.environ.get("LIBBSON_VERSION", "1.0.0")
+
 # Set a default value for MACOSX_DEPLOYMENT_TARGET.
 os.environ.setdefault("MACOSX_DEPLOYMENT_TARGET", "10.15")
 
@@ -40,7 +42,8 @@ def get_min_libbson_version():
 
 
 def append_libbson_flags(module):
-    pc_path = "libbson-1.0"
+    BSON_MAJOR_VERSION = int(LIBBSON_VERSION[0])
+    pc_path = "libbson-1.0" if BSON_MAJOR_VERSION == 1 else "bson2"
     install_dir = os.environ.get("LIBBSON_INSTALL_DIR")
     if install_dir:
         install_dir = os.path.abspath(install_dir)
@@ -48,11 +51,11 @@ def append_libbson_flags(module):
         # Handle the copy-able library file if applicable.
         if COPY_LIBBSON:
             if platform == "darwin":
-                lib_file = "libbson-1.0.0.dylib"
+                lib_file = "libbson-1.0.0.dylib" if BSON_MAJOR_VERSION == 1 else "bson2.dylib"
             elif platform == "linux":
-                lib_file = "libbson-1.0.so.0"
+                lib_file = "libbson-1.0.so.0" if BSON_MAJOR_VERSION == 1 else "bson2.so.0"
             else:  # windows
-                lib_file = "bson-1.0.dll"
+                lib_file = "bson-1.0.dll" if BSON_MAJOR_VERSION == 1 else "bson2.dll"
             lib_dir = "bin" if IS_WIN else "lib*"
             lib_dir = glob.glob(os.path.join(install_dir, lib_dir))
             if lib_dir:
@@ -81,16 +84,18 @@ def append_libbson_flags(module):
             if IS_WIN:
                 # Note: we replace any forward slashes with backslashes so the path
                 # can be parsed by bash.
-                lib_path = os.path.join(lib_dir, "bson-1.0.lib").replace(os.sep, "/")
+                bson_lib = "bson-1.0.lib" if BSON_MAJOR_VERSION == 1 else "bson2.lib"
+                lib_path = os.path.join(lib_dir, bson_lib).replace(os.sep, "/")
                 if os.path.exists(lib_path):
                     module.extra_link_args = [lib_path]
-                    include_dir = os.path.join(install_dir, "include", "libbson-1.0").replace(
+                    include_path = bson_lib.replace(".lib", "")
+                    include_dir = os.path.join(install_dir, "include", include_path).replace(
                         os.sep, "/"
                     )
                     module.include_dirs.append(include_dir)
                 else:
                     raise ValueError(f"Could not find the compiled libbson in {install_dir}")
-            pc_path = os.path.join(install_dir, lib_dir, "pkgconfig", "libbson-1.0.pc")
+            pc_path = os.path.join(install_dir, lib_dir, "pkgconfig", f"{pc_path}.pc")
 
     elif IS_WIN:
         raise ValueError("We require a LIBBSON_INSTALL_DIR with a compiled library on Windows")

